scripts/ensure_relative_imports.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is to ensure that relative imports exist within @streamlit/lib/dist
+# after yarn buildLibProd is done. This is important as if one is consuming @streamlit/lib/dist,
+# one will not have reference to @streamlit/lib/src and one actually needs relative pathing.
+
+import os
+import subprocess
+
+
+def check_for_string_in_files(file_paths, search_string):
+    for file_path in file_paths:
+        if not os.path.exists(file_path):
+            print(f"File not found: {file_path}")
+            exit(1)
+
+        with open(file_path, "r") as file:
+            content = file.read()
+            if search_string in content:
+                print(f"Found '{search_string}' in {file_path}")
+                exit(1)
+            else:
+                print(
+                    f"Success! Did not find '{search_string}' so we do have relative pathing in the lib/dist folder"
+                )
+
+
+def build_frontend_lib():
+    try:
+        subprocess.run(["yarn", "buildLibProd"], check=True, cwd="./frontend")
+    except subprocess.CalledProcessError as e:
+        print("Failed to build the necessary files.")
+        exit(1)
+
+
+if __name__ == "__main__":
+    base_dir = "./frontend/lib/dist/components/core/Block/"
+    block_js_path = os.path.join(base_dir, "Block.js")
+    block_d_ts_path = os.path.join(base_dir, "Block.d.ts")
+
+    search_string = "@streamlit/lib/src"
+
```

Comment:
As discussed. This doesn't really ensure anything. I think there are simpler solutions...commandline perhaps that would help address the problem. We should also consider precommit as a solution as well.

Reply:
I think we should not do precommit because when internal developers are developing, they won't be running the prod version. As a result, will be keeping it as a github actions. Let me know if you disagree.